source/Octopus.Client/Repositories/Async/UserRepository.cs

```diff
@@ -68,13 +68,14 @@ public async Task<UserResource> Register(RegisterCommand registerCommand)
             return await Client.Post<UserResource,UserResource>(Client.Link("Register"), registerCommand).ConfigureAwait(false);
         }
 
-        public Task SignIn(LoginCommand loginCommand)
+        public async Task SignIn(LoginCommand loginCommand)
         {
             if (loginCommand.State == null)
             {
                 loginCommand.State = new LoginState { UsingSecureConnection = Client.IsUsingSecureConnection };
             }
-            return Client.Post(Client.Link("SignIn"), loginCommand);
+            await Client.Post(Client.Link("SignIn"), loginCommand);
+            await Client.ReloadRootDocumentsAfterUserSignedIn();
```

Comment:
There is a use case where a client initialized as unauthenticated and then sign in later. As the client now will load space and user info which require authentication and permissions, spaceRootDocument is not populated for unauthenticated client and get reloaded after user signed in.